src/uws_client.c

```diff
@@ -202,6 +202,8 @@ UWS_CLIENT_HANDLE uws_client_create(const char* hostname, unsigned int port, con
                                     socketio_config.port = port;
                                     socketio_config.accepted_socket = NULL;
 
+                                    tlsio_config.hostname = hostname;
```

Comment:
Although it's understandable that the char pointer saved here would prevent more memory allocation, there is a risk here of bad memory management happening. 
I would recommending following the same practice within this function of copying the value (see mallocAndStrcpy_s above) instead of passing the pointer.

Reply:
I'm not entirely convinced this is actually the practice in this function.
A copy of the hostname is created on L148 and stored in the result structure.
However, whenever hostname is used as a part of a config to invoke xio_create, it's passed as is without creating an extra copy. See lines 201 and 233.

Reply:
Fine, we will fix it later.